docs/docs/tutorial/creating-a-source-plugin/part-7/index.mdx

```diff
@@ -0,0 +1,273 @@
+- The node manifests get written out in the `public` dir of your gatsby site, so you can check the manifests on your local disk at `/public/__node-manifests/<sourcePluginName>/<manifestId>.json` or you can navigate directly to that piece of content `https://<your-domain>/__node-manifests/<sourcePluginName>/<manifestId>`.
+
+## MIME types
+
+A media type (also known as a Multipurpose Internet Mail Extensions or MIME type) indicates the nature and format of a document, file, or assortment of bytes. Learn more about MIME types on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_Types).
+
+In [Part 2](/docs/tutorial/creating-a-source-plugin/part-2/#createnode) you learned how to use [`createNode`](/docs/reference/config-files/actions/#createNode). Inside the `internal` object that holds the `type` and `contentDigest` you can optionally also pass a `mediaType` key.
+
+The usage for a markdown file looks something like this:
+
+```ts
+createNode({
+  foo: `bar`,
+  id: `some-unique-id`,
+  parent: null,
+  children: [],
+  internal: {
+    type: `TypeName`,
+    // highlight-next-line
+    mediaType: `text/markdown`,
+    contentDigest: `contentDigest`
+  }
+})
+```
+
+Source plugins are responsible for setting the `mediaType` on their nodes (if necessary). By providing this key you tell transformer plugins (see [Creating a Transformer Plugin guide](/docs/how-to/plugins-and-themes/creating-a-transformer-plugin/)) that the node contains raw content that they can transform. Use either [official media types](https://www.iana.org/assignments/media-types/media-types.xhtml) or a made-up one if your data doesn't fit into any existing bucket.
+
+This distinction between Gatsby source plugins and transformer plugins is important as it allows both to be contained and focused on their specific job. Source plugins don't need to worry about transforming their data, transformer plugins don't care where the data is coming from.
+
+For example, the node above with the `text/markdown` MIME type will be transformed by `gatsby-transformer-remark` into further `MarkdownRemark` nodes.
+
+## Tracing
+
+If you want to give your users the ability to see tracing of your source plugin (e.g. in [OpenTelemetry](https://opentelemetry.io/) format) or use the performance tracing yourself to debug bottlenecks, be sure to read the [Performance Tracing guide](/docs/performance-tracing/). The guide has all the information you'd need.
+
+In context of source plugins, the [tracing](/docs/reference/config-files/node-api-helpers/#tracing) and [parentSpan](/docs/reference/config-files/node-api-helpers/#parentSpan) Node API helpers are relevant.
+
+When calling `startSpan` on the root tracer pass the `parentSpan` along, for example:
+
+```ts
+import opentracing from "opentracing"
```

Comment:
I wouldn't suggest to import `opentracing` like that to get access to `globalTracer()`, because there might be a mess with `node_modules` structure and this failing to work unless plugin has `opentracing` as actual dep, and then there might be multiple instaces of that package.

If you want `tracer` instance, you should grab `tracing.tracer` from first argument. `tracing.startSpan` takes care of setting `parentSpan` so that user doesn't have to use `childOf` (at least for their top level spans) so that's easiest way to use it:

```js
exports.sourceNodes = async ({ actions, tracing }) => {
  const span = tracing.startSpan(`foo`)

  // Perform any span operations. E.g. add a tag to your span
  span.setTag(`bar`, `baz`)

  // Rest of your plugin code

  span.finish()
}
```

or 

```js
exports.sourceNodes = async ({ actions, tracing: { tracer, parentSpan } }) => {
  const span = tracer.startSpan(`foo`, { childOf: parentSpan } )

  // Perform any span operations. E.g. add a tag to your span
  span.setTag(`bar`, `baz`)

  // Rest of your plugin code

  span.finish()
}
```

same `parentSpan` is also available as top level (as you have it in current example)